src/forms-opportunities/utils.js

```diff
@@ -200,23 +200,31 @@ export function filterForms(formOpportunities, scrapedData, log) {
 
   return formOpportunities.filter((opportunity) => {
     let urlMatches = false;
-    // Find matching form in scraped data
-    const matchingForm = scrapedData.formData.find((form) => {
+    let isSearchForm = false;
+
+    for (const form of scrapedData.formData) {
       const formUrl = new URL(form.finalUrl);
       const opportunityUrl = new URL(opportunity.form);
-      // eslint-disable-next-line max-len
-      urlMatches = opportunityUrl && formUrl.origin + formUrl.pathname === opportunityUrl.origin + opportunityUrl.pathname;
-
-      const isSearchForm = Array.isArray(form.scrapeResult)
-          && form.scrapeResult.some((result) => result?.formType === 'search' || result?.classList?.includes('search') || result?.classList?.includes('unsubscribe') || result?.action?.endsWith('search.html'));
 
-      return urlMatches && isSearchForm;
-    });
+      if (formUrl.origin + formUrl.pathname === opportunityUrl.origin + opportunityUrl.pathname) {
+        urlMatches = true;
+        isSearchForm = isNonEmptyArray(form.scrapeResult)
+            && form.scrapeResult.some((result) => result?.formType === 'search'
+                || result?.classList?.includes('search')
+                || result?.classList?.includes('unsubscribe')
+                || result?.action?.endsWith('search.html')
+                || (result?.fieldsLabels && isNonEmptyArray(result.fieldsLabels) && result.fieldsLabels.every((label) => label.toLowerCase().includes('search'))));
+
+        if (isSearchForm) {
+          break; // Stop looping once we find a match
```

Comment:
We eould need to improve this. Consider 2 forms in a page: one omis a search form and other is a contact  us form. We'd not like the opportunity to be filtered out in this case.